factories/settingsValidatorFactory.js

```diff
@@ -37,6 +37,16 @@
           args = { q: ['*:*'] };
         } else if ( self.searchEngine === 'es' || self.searchEngine === 'os') {
           fields = null;
+        } else if ( self.searchEngine === 'vectara') {
+          args = { query: [
+              {
+                query: '#$query##',
+                numResults: 10,
+                corpusKey :[{
+                  corpusId: 1
```

Comment:
;-)   Sigh.    We'll get there!   I wonder if we had a `#$query##`.

Reply:
Ideally, we would configure the Vectara-specific options right in a form in wizard, pass them in here and, yes, replace that corpusId with the one from that dialog.

Reply:
I wonder if the corpousId should be somethign like `corpusId: REPLACE_ME` ?